odoh.go

```diff
@@ -389,44 +336,55 @@ func CreateDefaultKeyPair() (ObliviousDoHKeyPair, error) {
 
 type QueryContext struct {
 	odohSecret []byte
-	suite      hpke.CipherSuite
+	suite      hpke.Suite
 	query      []byte
 	publicKey  ObliviousDoHConfigContents
 }
 
 func (c QueryContext) DecryptResponse(message ObliviousDNSMessage) ([]byte, error) {
 	aad := append([]byte{byte(ResponseType)}, []byte{0x00, 0x00}...) // 0-length encoded KeyID
 
-	odohPRK := c.suite.KDF.Extract(c.query, c.odohSecret)
-	key := c.suite.KDF.Expand(odohPRK, []byte(ODOH_LABEL_KEY), c.suite.AEAD.KeySize())
-	nonce := c.suite.KDF.Expand(odohPRK, []byte(ODOH_LABEL_NONCE), c.suite.AEAD.NonceSize())
+	h := c.suite.KdfID.Hash()
+	odohPRK := hkdf.Extract(h.New, c.query, c.odohSecret)
```

Comment:
I think we should revert this. The previous code uses the HPKE KDF interface directly, rather than the internal HKDF interface. If we ever wanted to support an HPKE suite that used a different KDF, this would require a change.

Reply:
I have exposed KDF API in https://github.com/armfazh/circl/pull/8